src/lib/capacityEngine.js

```diff
@@ -31,118 +34,6 @@ function getUtilizationRate(capacityAvailableNXM, capacityUsedNXM) {
   return capacityUsedNXM.mul(BASIS_POINTS).div(totalCapacity);
 }
 
-/**
- * Calculates available capacity for a pool.
- *
- * @param {Array<BigNumber>} trancheCapacities - Array of capacity BigNumbers.
- * @param {Array<BigNumber>} allocations - Array of allocation BigNumbers.
- * @param {number} firstUsableTrancheIndex - Index of the first usable tranche.
- * @returns {BigNumber} The available capacity as a BigNumber.
- */
-function calculateAvailableCapacity(trancheCapacities, allocations, firstUsableTrancheIndex) {
-  const unused = trancheCapacities.reduce((available, capacity, index) => {
-    const allocationDifference = capacity.sub(allocations[index]);
-    const allocationToAdd =
-      index < firstUsableTrancheIndex
-        ? bnMin(allocationDifference, Zero) // only carry over the negative
-        : allocationDifference;
-    return available.add(allocationToAdd);
-  }, Zero);
-  return bnMax(unused, Zero);
-}
```

Comment:
common functions was moved from *Engine.js files to helper.js to avoid circular dependencies